src/components/application_manager/src/commands/mobile/perform_audio_pass_thru_request.cc

```diff
@@ -362,7 +375,53 @@ void PerformAudioPassThruRequest::FinishTTSSpeak() {
 
 bool PerformAudioPassThruRequest::IsWaitingHMIResponse() {
   LOG4CXX_AUTO_TRACE(logger_);
-  return awaiting_tts_speak_response_ || awaiting_ui_response_;
+  return GetInterfaceAwaitState(HmiInterfaces::HMI_INTERFACE_TTS) ||
+         GetInterfaceAwaitState(HmiInterfaces::HMI_INTERFACE_UI);
+}
+
+void PerformAudioPassThruRequest::ProcessAudioPassThruIcon(
+    ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  smart_objects::SmartObject& msg_params = (*message_)[strings::msg_params];
+
+  if (msg_params.keyExists(strings::audio_pass_thru_icon)) {
```

Comment:
@dev-gh sometimes your pattern is applicable, but not always, and not in this certain case.